graph/appender/sidecars_check.go

```diff
@@ -44,33 +44,29 @@ func (a *SidecarsCheckAppender) applySidecarsChecks(trafficMap graph.TrafficMap,
 		}
 
 		// get the pods for the node, either by app+version labels, or workload deployment
-		var podLabels string
 		var err error
+		var pods *models.Pods
 		switch n.NodeType {
 		case graph.NodeTypeWorkload:
-			podLabels, err = a.getWorkloadLabels(n.Namespace, n.Workload, k8s)
-			if err != nil {
-				continue
+			workload, err := business.Workload.GetWorkload(n.Namespace, n.Workload, false)
+			if err == nil {
+				pods = &workload.Pods
 			}
 		case graph.NodeTypeApp:
-			podLabels = a.getAppLabels(appLabel, n.App, versionLabel, n.Version)
+			podLabels := a.getAppLabels(appLabel, n.App, versionLabel, n.Version)
+			pods, err = business.Workload.GetPods(n.Namespace, podLabels)
 		default:
 			continue
 		}
-		pods, err := k8s.GetPods(n.Namespace, podLabels)
 		checkError(err)
 
-		if len(pods.Items) == 0 {
-			log.Warningf("Sidecar check found no pods Checking sidecars node [%s] num pods [%v]", n.ID, len(pods.Items))
+		if pods == nil || (pods != nil && len(*pods) == 0) {
```

Comment:
I think it could be simpler if `pods` was directly a slice and not a pointer to a slice. Internally slice is already a pointer to an array (see https://blog.golang.org/go-slices-usage-and-internals , section "Slice internals"). So having `var pods *models.Pods` is like having a pointer to a pointer to an array.

Could be done simply by changing `business.Workload.GetWorkload(...)` to return `models.Pods`

Reply:
+1 Yes, I will refactor this one.